compiler/rustc_middle/src/ty/print/pretty.rs

```diff
@@ -2124,6 +2124,7 @@ fn for_each_def(tcx: TyCtxt<'_>, mut collect_fn: impl for<'b> FnMut(&'b Ident, N
 
             match child.res {
                 def::Res::Def(DefKind::AssocTy, _) => {}
+                def::Res::Def(DefKind::TyAlias, _) => {}
```

Comment:
Just to make sure that I'm understanding this right: 

This PR modifies `for_each_def` so that `collect_fn` isn't invoked for type aliases, therefore `trimmed_def_paths` won't find both `std::io::Result` and `std::result::Result` (for example, you don't hardcode these types but they're the natural example) having the same name because `std::io::Result` is a type alias, which will let `Result` be shortened.

Reply:
That's correct.